src/components/TodoModal/TodoModal.tsx

```diff
@@ -15,25 +34,32 @@ export const TodoModal: React.FC = () => {
               className="modal-card-title has-text-weight-medium"
               data-cy="modal-header"
             >
-              Todo #2
+              {`Todo #${todo?.id}`}
```

Comment:
Null Safety: Ensure that `todo` is not `null` before accessing its `id` property to avoid rendering issues.